utils/lib/rr_graph/channel2.py

```diff
@@ -53,38 +52,74 @@ def __init__(self, tracks):
         self.trees = []
         self.tracks = sorted(tracks, key=lambda x: x[1] - x[0])
 
-    def _place_track(self, track):
-        """Add track to existing interval tree, if there is room (ie doesn't overlap)
-        If track won't fit in any existing tree, allocate a new tree for the track"""
-        for idx, tree in enumerate(self.trees):
-            if not tree.overlaps(track[0], track[1] + 1):
-                tree.add(
-                    Interval(begin=track[0], end=track[1] + 1, data=track[2])
-                )
-                return
+    def _start_track(self, track):
+        self.trees.append(list())
+        self.trees[-1].append(track)
 
-        self.trees.append(IntervalTree())
-        self.trees[-1].add(
-            Interval(begin=track[0], end=track[1] + 1, data=track[2])
-        )
+    def _add_track_to_last_tree(self, track, idx=-1):
+        self.trees[idx].append(track)
```

Comment:
Is the last tree `self.trees[-1]`? Because the name seems misleading if supplied with an `idx` argument.

Reply:
The function name is wrong, I'll update.